src/options/output.rs

```diff
@@ -152,53 +147,136 @@ impl OutputStrategy {
         }
     }
 
-    pub fn get_strategy(&self) -> (FormatType, FinalOutputType) {
-        let output_type = self.output_type.clone();
+    /// Handle output and errors from a Rover command.
+    pub fn handle_output<T>(&self, rover_command_output: T) -> RoverResult<()>
+    where
+        T: RoverPrinter,
+    {
+        rover_command_output.write_or_print(self)
+    }
+
+    /// Get the format (plain/json) and strategy (stdout/file)
+    pub fn get_format_and_strategy(&self) -> (RoverOutputFormatKind, RoverOutputDestination) {
```

Comment:
factored this out into its own function so it could be reused in multiple places